pkg/backend/openshiftcluster.go

```diff
@@ -197,19 +198,41 @@ func (ocb *openShiftClusterBackend) updateAsyncOperation(ctx context.Context, lo
 			now := time.Now()
 			asyncdoc.AsyncOperation.EndTime = &now
 
+			// Returns a generic error
+			handleOtherError := func() {
+				log.Error(backendErr)
+				asyncdoc.AsyncOperation.Error = &api.CloudErrorBody{
+					Code:    api.CloudErrorCodeInternalServerError,
+					Message: "Internal server error.",
+				}
+			}
+
 			if provisioningState == api.ProvisioningStateFailed {
-				// if type is CloudError - we want to propagate it to the
-				// asyncOperations errors. Otherwise - return generic error
-				err, ok := backendErr.(*api.CloudError)
-				if ok {
+				switch err := backendErr.(type) {
+				case *api.CloudError:
+					// if type is CloudError - we want to propagate it to the asyncOperations errors.
 					log.Print(backendErr)
 					asyncdoc.AsyncOperation.Error = err.CloudErrorBody
-				} else {
-					log.Error(backendErr)
+				case *azure.RequestError:
+					// if it is RequestDisallowedByPolicy, tell user so they can change policy
+					if err.ServiceError != nil && err.ServiceError.Code == api.CloudErrorRequestDisallowedByPolicy {
+						log.Print(backendErr)
+						asyncdoc.AsyncOperation.Error = &api.CloudErrorBody{
+							Code:    api.CloudErrorRequestDisallowedByPolicy,
+							Message: err.ServiceError.Message,
+						}
+					} else {
+						handleOtherError()
+					}
+				case adal.TokenRefreshError:
+					// if it is a TokenRefreshError, user should be given info to troubleshoot AAD or other authentication issues
+					log.Print(backendErr)
 					asyncdoc.AsyncOperation.Error = &api.CloudErrorBody{
-						Code:    api.CloudErrorCodeInternalServerError,
-						Message: "Internal server error.",
+						Code:    api.CloudErrorTokenRefreshError,
+						Message: err.Error(),
 					}
+				default:
+					handleOtherError()
```

Comment:
Will this enables customer to see ALL other errors?  I don't think this is desired behaviour.